modules/serverless-invoking-go-functions.adoc

```diff
@@ -0,0 +1,99 @@
+// Module included in the following assemblies
+//
+// * /serverless/functions/serverless-developing-go-functions.adoc
+
+[id="serverless-invoking-go-functions_{context}"]
+= About invoking Golang functions
+
+Golang functions are invoked by using different methods, depending on whether they are triggered by a HTTP request or a CloudEvent.
+
+[id="serverless-invoking-go-functions-http_{context}"]
+== Function triggered by HTTP request
+
+When an incoming HTTP request is received, your function will be invoked with a standard Golang link:https://golang.org/pkg/context/[Context] as the first parameter, followed by two more parameters:
+
+* link:https://golang.org/pkg/net/http/#ResponseWriter[`http.ResponseWriter`]
+* link:https://golang.org/pkg/net/http/#Request[`http.Request`]
+
+You can use standard Golang techniques to access the request, and set a proper HTTP response of your function.
+
+.Example HTTP response
+[source,go]
+----
+func Handle(ctx context.Context, res http.ResponseWriter, req *http.Request) {
+  // Read body
+  body, err := ioutil.ReadAll(req.Body)
+  defer req.Body.Close()
+  if err != nil {
+	http.Error(res, err.Error(), 500)
+	return
+  }
+  // Process body and function logic
+  // ...
+}
+----
+
+[id="serverless-invoking-go-functions-cloudevent_{context}"]
+== Function triggered by CloudEvent
+
+When an incoming CloudEvent is received, the event is invoked by the link:https://cloudevents.github.io/sdk-go/[CloudEvents Golang SDK] and the `Event` type as a parameter.
+
+You can leverage the Golang link:https://golang.org/pkg/context/[Context] as an optional parameter in the function contract, as shown in the list of supported function signatures:
+
+.Supported function signatures
+[source,go]
+----
+Handle()
+Handle() error
+Handle(context.Context)
+Handle(context.Context) error
+Handle(cloudevents.Event)
+Handle(cloudevents.Event) error
+Handle(context.Context, cloudevents.Event)
+Handle(context.Context, cloudevents.Event) error
+Handle(cloudevents.Event) *cloudevents.Event
+Handle(cloudevents.Event) (*cloudevents.Event, error)
+Handle(context.Context, cloudevents.Event) *cloudevents.Event
+Handle(context.Context, cloudevents.Event) (*cloudevents.Event, error)
+----
+
+.CloudEvent example
+
+. A CloudEvent is received which contains a JSON string in its data property:
```

Comment:
At first glance to me, this numbering kind of made it look like this should be a procedure module. I can't tell if this is really trying to go with an example of a flow, or if it really should be a procedure module, because it is telling you options of how to do something.

I'd say to take another look to see if a procedure module makes sense. At the very least, I'd suggest to add an intro sentence into the steps (perhaps in place of the "CloudEvent example" heading).

Reply:
Removed the numbering since this seems like more of a walkthrough example?